R/resolver.R

```diff
@@ -352,64 +315,98 @@ resolve_vector_l <- function(expr, vector) {
       expr = rlang::eval_tidy(expr = quo, data = NULL)
     )
 
+  item_names <- vector
+  item_count <- length(vector)
+  item_sequence <- seq_along(vector)
+
+  resolved <-
+    perform_resolution(
+      resolved = resolved,
+      item = item,
+      item_count = item_count,
+      item_names = item_names,
+      item_sequence = item_sequence
+    )
+
+  resolved
+}
+
+resolve_vector_i <- function(expr, vector, item = "item") {
+  which(resolve_vector_l(expr = {{ expr }}, vector = vector, item = item))
+}
+
+resolver_stop_on_logical <- function(item) {
+
+  stop(
+    "The number of logical values must either be 1 or the number of ",
+    item, "s",
+    call. = FALSE
+  )
+}
+
+resolver_stop_on_numeric <- function(item, unknown_resolved) {
+
+  stop(
+    "The following ", item ," indices do not exist in the data: ",
+    paste0(unknown_resolved, collapse = ", "),
+    call. = FALSE
+  )
+}
+
+resolver_stop_on_character <- function(item, unknown_resolved) {
+
+  stop(
+    "The following ", item , "(s) do not exist in the data: ",
+    paste0(unknown_resolved, collapse = ", "),
+    call. = FALSE
+  )
+}
+
+resolver_stop_unknown <- function(item, resolved) {
+
+  stop(
+    "Don't know how to select ", item , "s using an object of class ",
+    class(resolved)[1],
+    call. = FALSE
+  )
+}
+
+perform_resolution <- function(resolved, item, item_count, item_names, item_sequence) {
```

Comment:
1. Let's rename `perform_resolution` to `normalize_resolved`
2. Let's derive `item_count` and `item_sequence` from `item_names` as needed, instead of taking them all as arguments
3. Let's rename `item` to `item_label`

Reply:
All changed now.